lib/monaco-adapter.js

```diff
@@ -272,7 +272,7 @@ var MonacoAdapter = function () {
    */
   MonacoAdapter.prototype.registerUndo = function registerUndo(callback) {
     if (typeof callback === 'function') {
-      this.callbacks.undo = callback;
```

Comment:
`registerUndo` is called once during initialisation, but internally monaco can change model. This is required to handle such scenario. Rest looks fine. Thank you!

Reply:
I don't think my approach is right as it can go out of sync and generate errors. Can you think of a way to solve this?

Reply:
I am yet to test this out. I can look again then.

Reply:
@Progyan1997 @adamjimenez just let me know when you two are satisfied with the solution here, this is beyond my knowledge of how Monaco works so I will defer to you.